File: taskhub-web/src/services/auth.service.ts
import axios, { AxiosError } from "axios";

import { BaseResponse } from "@/types/base-response.type";
import {
  LoginPayload,
  LoginData,
  RegisterPayload,
  RegisterData,
} from "@/types/auth.type";


const api = axios.create({
  baseURL: `${process.env.NEXT_PUBLIC_API_URL}/auth`,
  headers: { "Content-Type": "application/json" },
});


export async function login(
  payload: LoginPayload
): Promise<BaseResponse<LoginData>> {
  try {
    const res = await api.post("/login", payload);
    return res.data;
  } catch (err) {
   const error = err as AxiosError<{ message: string }>;
    throw new Error(error.response?.data?.message || "Register failed");
  }
}

export async function register(
  payload: RegisterPayload
): Promise<BaseResponse<RegisterData>> {
  try {
    const res = await api.post("/register", payload);
    return res.data;
  } catch (err) {
   const error = err as AxiosError<{ message: string }>;
    throw new Error(error.response?.data?.message || "Register failed");
  }
}

File: taskhub_mobile/lib/data/services/auth_service.dart
import 'dart:convert';
import 'package:flutter_dotenv/flutter_dotenv.dart';
import 'package:http/http.dart' as http;
import 'package:taskhub_mobile/data/models/base_response_model.dart';
import 'package:taskhub_mobile/data/models/auth_model.dart';

class AuthService {
  static final String? _baseUrl = dotenv.env['API_URL'];
  static final _client = http.Client();

  static Future<BaseResponse<LoginData>> login(LoginPayload payload) async {
    final url = Uri.parse('$_baseUrl/auth/login');

    final response = await _client.post(
      url,
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode(payload.toJson()),
    );

    final json = jsonDecode(response.body);

    if (response.statusCode == 200) {
      return BaseResponse.fromJson(json, (data) => LoginData.fromJson(data));
    } else {
      throw Exception(json['message'] ?? 'Login failed');
    }
  }

  static Future<BaseResponse<RegisterData>> register(
    RegisterPayload payload,
  ) async {
    final url = Uri.parse('$_baseUrl/auth/register');

    final response = await _client.post(
      url,
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode(payload.toJson()),
    );

    final json = jsonDecode(response.body);

    if (response.statusCode == 200) {
      return BaseResponse.fromJson(json, (data) => RegisterData.fromJson(data));
    } else {
      throw Exception(json['message'] ?? 'Register failed');
    }
  }
}
